server/src/main/java/org/candlepin/policy/js/pool/PoolRules.java

```diff
@@ -555,6 +555,21 @@ private void syncBranding(Pool pool, Pool existingPool) {
         return incomingProvided;
     }
 
+    private boolean changedProductsInSet(Set<Product> products,
+        Map<String, Product> changedProducts) {
+
+        if (products != null && changedProducts != null) {
+            for (Product product: products) {
+                if (product != null && product.getId() != null &&
```

Comment:
You can skip the null check on the ID here. If it's null, it won't be found in the products list, nor will it cause an error.

Reply:
im only 50% sure on this, but during debugging I *thought* I saw some null objects. I am reluctant to try and prove it, may I ask what is the harm in an additional check or two even if that is not the case?